Add deprecation warnings, Use non-deprecated upstream methods

j9ac9k opened the pull request:
In many portions of the codebase, there are commented out sections about methods or classes, or key-word arguments being deprecated, but they do not emit actual deprecation warnings.

This PR makes it so the use of deprecated methods emit deprecation warnings.  Furthermore, the code-base, examples and test-suite have been altered so that the deprecated methods are not used (exception in the test suite, where some tests test those explicitly).

This PR also restore the public API bits whose removal caused #1535  (restored, but added a deprecation warning).

Furthermore, thanks to @swt2c we got a heads up on issues involving Python 3.10, this PR addresses those as well.

Fixes #1535 
Fixes #1571

swt2c commented on examples/relativity/relativity.py:
```diff
@@ -131,7 +131,7 @@ def recalculate(self):
         if a:
             self.lastAnimTime = pg.ptime.time()
-            self.animTimer.start(self.animDt*1000)
+            self.animTimer.start((self.animDt*1000))
```
Should this be `self.animTimer.start(int(self.animDt*1000))` ?

j9ac9k replied:
absolutely should be; not sure how I messed that up; I redid that bit of git history to save myself the embarrassment

ksunden commented on pyqtgraph/Qt.py:
```diff
@@ -290,6 +290,59 @@ def pyqt5_qabort_override(*args, **kwds):
+                tr.scale(sx, sy)	
+                self.setTransform(tr)	
+            else:	
+                return __QGraphicsItem_scale(self)
```
should this return? weird that the other code path has no return but this does...

pijyoi replied:
Yes it does. ```scale()``` shadows a Qt5 _getter_ method.

ksunden commented on pyqtgraph/__init__.py:
```diff
@@ -384,29 +384,18 @@ def exit():
-    #if len(args)+len(kargs) > 0:
-        #w.plot(*args, **kargs)
-        
+    mkQApp()      
```
trailing whitespace

ksunden commented on pyqtgraph/functions.py:
```diff
@@ -1129,7 +1129,7 @@ def makeARGB(data, lut=None, levels=None, scale=None, useRGBA=False, output=None
     if not isinstance(levels, xp.ndarray):
         levels = xp.array(levels)
-    levels = levels.astype(xp.float)
+    levels = levels.astype(xp.float32)
```
why 32, not 64?

j9ac9k replied:
There was discussion about that here: https://github.com/pyqtgraph/pyqtgraph/pull/1518#issuecomment-769602093

pijyoi replied:
The discussion was concerning the choice of image data used for benchmarking. Users are still free to supply float64 images. So in this case, casting levels to float32 unconditionally might not be safe?

j9ac9k replied:
Fix added to #1577

ksunden commented on pyqtgraph/graphicsItems/PlotDataItem.py:
```diff
@@ -9,6 +9,7 @@
 from .. import functions as fn
 from .. import debug as debug
 from .. import getConfigOption
+import warnings
```
already imported

j9ac9k replied:
Fixed in #1577

ksunden commented on pyqtgraph/widgets/SpinBox.py:
```diff
@@ -77,7 +77,6 @@ def __init__(self, parent=None, value=0.0, **kwargs):
             'step': D('0.01'),  ## if 'dec' is false, the spinBox steps by 'step' every time
                                 ## if 'dec' is True, the step size is relative to the value
                                 ## 'step' needs to be an integral divisor of ten, ie 'step'*n=10 for some integer value of n (but only if dec is True)
-            'log': False,   # deprecated
```
Is there a reason this one is not warned like the others?

j9ac9k replied:
It would only raise an Exception

```python
            if self.opts['log']:	
                raise Exception("Log mode no longer supported.")
```

Figure I'm just removing reference in the documentation, I'm not actually removing any functionality.

ksunden commented:
Just realized that I had an old gh notifications window up, so I was reading through this after it was merged... nothing to serious, but a few comments if any of them should be addressed, happy to open a followup.

j9ac9k commented:
@ksunden Thanks for taking a look at this, those were some sloppy mistakes, I've addressed your comments, or explained them in the comments.  I'll create a new PR momentarily with the needed changes.

j9ac9k commented:
Oh right, you had that handy numpy method to see if you could cast as
float32, I should probably use that here too.

On Sun, Feb 14, 2021 at 14:32 pijyoi <notifications@github.com> wrote:

> *@pijyoi* commented on this pull request.
> ------------------------------
>
> In pyqtgraph/functions.py
> <https://github.com/pyqtgraph/pyqtgraph/pull/1573#discussion_r575874912>:
>
> > @@ -1129,7 +1129,7 @@ def makeARGB(data, lut=None, levels=None, scale=None, useRGBA=False, output=None
>              raise Exception('levels argument is required for float input types')
>      if not isinstance(levels, xp.ndarray):
>          levels = xp.array(levels)
> -    levels = levels.astype(xp.float)
> +    levels = levels.astype(xp.float32)
>
> The discussion was concerning the choice of image data used for
> benchmarking. Users are still free to supply float64 images. So in this
> case, casting levels to float32 unconditionally might not be safe?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/pyqtgraph/pyqtgraph/pull/1573#discussion_r575874912>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAE5Z7QUYIL2STD6EPHNMO3S7BFPJANCNFSM4XRNAJFA>
> .
>
